### PR TITLE
Migrate remaining controllers from requery to room

### PR DESCRIPTION
**Pull request by AndyScherzinger**
This PR migrates the remaining controllers from requery to room

* server selection
* web view login
* account verification
* location picker
* cleanup ChatController & GeocodingController

### CODE REVIEW

File: app/src/main/java/com/nextcloud/talk/controllers/WebViewLoginController.kt [hunk 1]
```diff
@@ -327,15 +325,16 @@ class WebViewLoginController(args: Bundle? = null) : NewBaseController(
         val loginData = parseLoginData(assembledPrefix, dataString)
         if (loginData != null) {
             dispose()
-            val currentUser = userUtils.currentUser
+            val currentUser = userManager.currentUser.blockingGet()
             var messageType: ApplicationWideMessageHolder.MessageType? = null
-            if (!isPasswordUpdate && userUtils.getIfUserWithUsernameAndServer(loginData.username, baseUrl)) {
+            if (!isPasswordUpdate &&
+                userManager.getIfUserWithUsernameAndServer(loginData.username!!, baseUrl!!).blockingGet()
```
[the remark this answers is not in the document]
Comment (AlvaroBrey) [reply app/src/main/java/com/nextcloud/talk/controllers/WebViewLoginController.kt:331]: If it _always_ needs to emit a value (`true` or `false` in this case), then it should not be a `Maybe`, but a `Single` instead.
Here's how to do that:

```kotlin
    fun getIfUserWithUsernameAndServer(username: String, server: String): Single<Boolean> {
        return userRepository
            .getUserWithUsernameAndServer(username, server)
            .map { true }
            .switchIfEmpty(Single.just(false))
    }
```

**Comment by AndyScherzinger** [comment]
@AlvaroBrey currentUser being null has been fixed with my latest commit, in case you have some time to spare :) I did test adding 1 and 2 users to a fresh installation and that worked as well as user switching

File: app/src/main/java/com/nextcloud/talk/utils/AccountUtils.kt [hunk 2]
```diff
@@ -1,7 +1,9 @@
 /*
  * Nextcloud Talk application
  *
+ * @author Andy Scherzinger
  * @author Mario Danic
+ * Copyright (C) 2021 Andy Scherzinger <info@andy-scherzinger.de>
```
Comment (AlvaroBrey) [inline app/src/main/java/com/nextcloud/talk/utils/AccountUtils.kt:6]: Wrong year!

**Review by AlvaroBrey — APPROVED** [review]

File: app/src/main/java/com/nextcloud/talk/utils/AccountUtils.kt (hunk 2 above)
Comment (AndyScherzinger) [reply app/src/main/java/com/nextcloud/talk/utils/AccountUtils.kt]: Fixed :)

**Pull request merged**